packages/hurl/src/runner/predicate.rs

```diff
@@ -603,6 +608,50 @@ fn eval_is_number(actual: &Value) -> Result<AssertResult, RunnerError> {
     })
 }
 
+/// Evaluates if an `actual` value is an IPv4 address.
+fn eval_is_ipv4(source_info: SourceInfo, actual: &Value) -> Result<AssertResult, RunnerError> {
+    let ip = actual.to_string();
+    let parsed = match IpAddr::from_str(&ip) {
+        Ok(ip) => ip,
+        Err(_) => {
+            return Err(RunnerError::new(
+                source_info,
+                RunnerErrorKind::InvalidIpAddr { value: ip },
+                false,
+            ))
+        }
+    };
+
+    Ok(AssertResult {
+        success: parsed.is_ipv4(),
+        actual: actual.repr(),
+        expected: "ipv4".to_string(),
+        type_mismatch: false,
+    })
+}
+
+/// Evaluates if an `actual` value is an IPv6 address.
+fn eval_is_ipv6(source_info: SourceInfo, actual: &Value) -> Result<AssertResult, RunnerError> {
+    let ip = actual.to_string();
+    let parsed = match IpAddr::from_str(&ip) {
+        Ok(ip) => ip,
+        Err(_) => {
+            return Err(RunnerError::new(
+                source_info,
+                RunnerErrorKind::InvalidIpAddr { value: ip },
```

Comment:
I think we can avoid creating the variant `InvalidIpAddr`. If a value is not parsable to an IP address, evaluating `isIpv6` on it should return `false` and not raise an error:

```
"Toto" isIpv6 == false
```